### PR TITLE
Updating and checking EKP examples

### PR DESCRIPTION
**Pull request by odunbar**
# Purpose 
To run through all current EKP examples, checking they run, checking whether manifests could be updated, and applying our new code-styles and interfaces. 

## Content
Checked and updated examples and docs for:
- [x] Loss Minimization
- [x] Loss Minimization Sparse
- [x] Lorenz (EKI/UKI/distributed) - actually uses priors with transformations, and produces gifs rather than pngs - per - iteration.
- [x] Localization
- [x] Cloudy(EKI/UKI) - also now produces gifs rather than displaying pngs
- [x] AerosolActivation - produces PDFs, and `CloudyMicrophysics` version is pinned to `v0.5`
- [x] ClimateMachine - updated priors

### CODE REVIEW

File: examples/AerosolActivation/aerosol_activation.jl [hunk 1]
```diff
@@ -70,25 +71,14 @@ osmotic_coeff_true = 0.9
...
-prior1 = Dict("distribution" => d1, "constraint" => c1, "name" => parameter_names[1])
-prior2 = Dict("distribution" => d2, "constraint" => c2, "name" => parameter_names[2])
-priors = EKP.ParameterDistributions.ParameterDistribution([prior1, prior2])
+# We must define parameter priors. Both parameters have to
+# be positive definite, therefore we define the constraints to
+# be bounded below by zero.  We don't have much other prior knowledge
+# about the parameters. We simply constrain their scale to be
+# loosely of size 1. For more details see [`constrained_gaussian`](@ref constrained-gaussian)
+prior1 = constrained_gaussian(parameter_names[1], 1.0, 1.0, 0.0, Inf)
+prior2 = constrained_gaussian(parameter_names[2], 1.0, 1.0, 0.0, Inf)
```
Comment (ilopezgp) [inline examples/AerosolActivation/aerosol_activation.jl:80]:
```suggestion
prior1 = constrained_gaussian(parameter_names[1], 1, 1, 0, Inf)
prior2 = constrained_gaussian(parameter_names[2], 1, 1, 0, Inf)
```
for readability

File: examples/AerosolActivation/aerosol_activation.jl [hunk 2]
```diff
@@ -206,11 +192,11 @@ function plot_ensemble_scatter(id)
 
     if id == 1
         ylabel = "Molar mass [kg/mol]"
-        filename = "molar_mass_scatter.svg"
+        filename = "molar_mass_scatter.pdf"
     end
     if id == 2
```
Comment (ilopezgp) [inline examples/AerosolActivation/aerosol_activation.jl:197]:
```suggestion
    elseif id == 2 
```

File: examples/Cloudy/Cloudy_example_uki.jl [hunk 3]
```diff
@@ -61,34 +62,12 @@ dist_true = ParticleDistributions.GammaPrimitiveParticleDistribution(ϕ_true...)
...
-
-priors = ParameterDistribution([prior_N0, prior_θ, prior_k])
-prior_mean = [4.5; 0.0; -1.0]
-prior_cov = Array(Diagonal(ones(Float64, length(prior_mean))))
+prior_N0 = constrained_gaussian(par_names[1], 200, 200, 0.4 * N0_true, Inf)
+prior_θ = constrained_gaussian(par_names[2], 1.0, 3.0, 1e-1, Inf)
+prior_k = constrained_gaussian(par_names[3], 0.2, 0.5, 1e-4, Inf)
+priors = combine_distributions([prior_N0, prior_θ, prior_k])
+#prior_mean = [4.5; 0.0; -1.0]
+#prior_cov = Array(Diagonal(ones(Float64, length(prior_mean))))
```
Comment (ilopezgp) [inline examples/Cloudy/Cloudy_example_uki.jl:70]: delete?

File: examples/Lorenz/Lorenz_example_ukp.jl [hunk 4]
```diff
@@ -70,26 +70,25 @@ println(params_true)
...
 end
 
+# if dynamics == 2
+#     prior_mean = [F_true + 1.0; A_true + 0.5]
+#     # prior_stds = [2.0, 0.5*A_true]
+#     prior_cov = Array(Diagonal((0.1 * prior_mean) .^ 2))
+# else
+#     prior_mean = [F_true;]
+#     prior_cov = ones(Float64, 1, 1)
+# end
```
Comment (ilopezgp) [inline examples/Lorenz/Lorenz_example_ukp.jl:90]: delete?

**Review by ilopezgp — APPROVED** [review]
LGTM after the changes!

File: docs/src/examples/Cloudy_example.md [hunk 5]
```diff
@@ -134,9 +120,9 @@ truth = Observations.Observation(y_t, Γy, data_names)
...
+* `Cloudy_example_uki.jl`: In addition to a point estimate of the optimal parameter (which is again given by the ensemble mean of the last iteration and printed to standard output), Unscented Kalman inversion also provides a covariance approximation of the posterior distribution. Together, the mean and covariance allow for the reconstruction of a Gaussian approximation of the posterior distribution. The evolution of this Gaussian approximation over subsequent iterations is shown as an animation over the computational (unconstrained) space. All parameters as well as the model output from the unscented Kalman inversion are stored in an output directory, as `parameter_storage_uki.jld2` and `data_storage_uki.jld2`.
```
[the remark this answers is not in the document]
Comment (odunbar) [reply docs/src/examples/Cloudy_example.md:125]: Done

File: examples/AerosolActivation/aerosol_activation.jl (hunk 1 above)
Comment (odunbar) [reply examples/AerosolActivation/aerosol_activation.jl:80]: Done

File: examples/AerosolActivation/aerosol_activation.jl (hunk 2 above)
Comment (odunbar) [reply examples/AerosolActivation/aerosol_activation.jl:197]: Done

File: examples/Cloudy/Cloudy_example_uki.jl (hunk 3 above)
Comment (odunbar) [reply examples/Cloudy/Cloudy_example_uki.jl:70]: Done

File: examples/Lorenz/Lorenz_example_ukp.jl (hunk 4 above)
Comment (odunbar) [reply examples/Lorenz/Lorenz_example_ukp.jl:90]: Done, and removed a bunch of other unnecessary comments

File: docs/src/examples/lorenz_example.md [hunk 6]
```diff
@@ -107,20 +104,20 @@ The L96 parameter estimation can be run using `julia --project Lorenz_example.jl
 
 
 ## Solution and Output
-The output will provide the estimated parameters.
+The output will provide the estimated parameters in the constrained `ϕ`-space. The `priors` is required in the get-method to apply these constraints.
```
[the remark this answers is not in the document]
Comment (odunbar) [reply docs/src/examples/lorenz_example.md]: Done

**Pull request merged**